shell/platform/darwin/macos/framework/Source/FlutterSurfaceManager.mm

```diff
@@ -82,8 +80,81 @@ - (void)swapBuffers {
             _frameBuffers[kFlutterSurfaceManagerFrontBuffer]);
 }
 
-- (uint32_t)glFrameBufferId {
-  return [_frameBuffers[kFlutterSurfaceManagerBackBuffer] glFrameBufferId];
+- (FlutterBackingStoreDescriptor*)renderBufferDescriptor {
+  uint32_t fboId = [_frameBuffers[kFlutterSurfaceManagerBackBuffer] glFrameBufferId];
+  return [[FlutterBackingStoreDescriptor alloc] initOpenGLDescriptorWithFBOId:fboId];
+}
+
+@end
+
+@implementation FlutterMetalSurfaceManager {
+  CGSize _surfaceSize;
+  id<MTLDevice> _mtlDevice;
+  id<MTLCommandQueue> _mtlCommandQueue;
+  CAMetalLayer* _containingLayer;  // provided (parent layer)
+  CALayer* _contentLayer;
+
+  FlutterIOSurfaceHolder* _ioSurfaces[kFlutterSurfaceManagerBufferCount];
+  id<MTLTexture> _textures[kFlutterSurfaceManagerBufferCount];
+}
+
+- (nullable instancetype)initWithDevice:(nonnull id<MTLDevice>)device
+                           commandQueue:(nonnull id<MTLCommandQueue>)commandQueue
+                             metalLayer:(nonnull CAMetalLayer*)layer {
+  self = [super init];
+  if (self) {
+    _mtlDevice = device;
+    _mtlCommandQueue = commandQueue;
+    _containingLayer = layer;
+
+    // Layer for content. This is separate from provided layer, because it needs to be flipped
+    // vertically if we render to Metal texture
+    _contentLayer = [[CALayer alloc] init];
+    [_containingLayer addSublayer:_contentLayer];
+
+    _ioSurfaces[0] = [FlutterIOSurfaceHolder alloc];
```

Comment:
Missing `init`.

Reply:
hah, not sure why it worked without it. Will fix it.

Reply:
`alloc` just allocates the memory, `init` calls the constructor. Maybe the `init` of the object and its base classes was empty?

Reply:
Yeah. I've added `init` to this now and also fixed another place where there was the same issue.